uniffi_meta/src/types.rs

```diff
@@ -143,6 +144,41 @@ impl Type {
         };
         Box::new(std::iter::once(self).chain(nested_types))
     }
+
+    pub fn name(&self) -> Option<String> {
+        match self {
+            Type::Object { name, .. } => Some(name.to_string()),
+            Type::Record { name, .. } => Some(name.to_string()),
+            Type::Enum { name, .. } => Some(name.to_string()),
+            Type::CallbackInterface { name, .. } => Some(name.to_string()),
+            Type::Custom { name, .. } => Some(name.to_string()),
+            Type::External { name, .. } => Some(name.to_string()),
+            Type::Optional { inner_type } | Type::Sequence { inner_type } => {
+                if inner_type.name().is_some() {
+                    Some(inner_type.name().unwrap())
+                } else {
+                    None
+                }
+            }
+            Type::Map { value_type, .. } => {
+                if value_type.name().is_some() {
+                    Some(value_type.name().unwrap())
+                } else {
+                    None
+                }
+            }
+            _ => None,
+        }
+    }
+
+    // Currently we just change the `import_name`. Different languages have slightly different
+    // naming patterns. We don't change the name of the external type, because the original
+    // names are used to find them in the `ComponentInterface`.
```

Comment:
This is a bit surprising to me.  I would think that we need to update the names for the types as well.  Maybe this is currently working because both Rust and Python use UpperCamelCase.  What if you updated the `class_name` function so that prepended `Py` to all classes or something like that to test that the type names would line up if the language bindings changed them.

My guess is that you would need to translate the `name` field for all types and if you did that, then you wouldn't need to add the `Type::External::import_name` field.